helix-core/src/main/java/org/apache/helix/api/rebalancer/constraint/dataprovider/PartitionQuotaProvider.java

```diff
@@ -0,0 +1,33 @@
+package org.apache.helix.api.rebalancer.constraint.dataprovider;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * An interface for getting partition quota information.
+ */
+public interface PartitionQuotaProvider {
+
+  /**
+   * @param resource
+   * @param partition
+   * @return The quota that is required by a certain partition in the specified resource
+   */
+  int getPartitionQuota(String resource, String partition);
```

Comment:
It is.
I'm struggling to make the choice between weight or quota. I named it quota to avoid confusing with the "constraint weight".

Given we named the constraint weight to importance, let's name it to weight here.

Reply:
Clarify it with comments int he javadoc, with examples. I prefer weight rather than quota

A specific example class with a lot of javadocs on different scenarios and how these classes are to be used, will not hurt. Tests can serve as examples, but they are usually sloppy, and if they are examples, then they tend to be copy-pasted the same way in real code.